docs/tutorials/jokes.md

```diff
@@ -1542,32 +1542,30 @@ This works just fine, but the problem is, during development, we don't want to c
 
 So we've got a bit of extra work to do to avoid this development time problem.
 
-Note that this isn't a remix-only problem. Any time you have "live reload" of server code, you're going to have to either disconnect and reconnect to databases (which can be slow) or do the workaround I'm about to show you.
+Note that this isn't a remix-only problem. Any time you have "live reload" of server code, you're going to have to either disconnect and reconnect to databases (which can be slow) or use the [`global` singleton workaround][global-singleton-workaround].
 
-💿 Copy this into a new file called `app/utils/db.server.ts`
+💿 Copy this into two new files called `app/utils/singleton.server.ts` & `app/utils/db.server.ts`
+
+```ts filename=app/utils/singleton.server.ts
+export const singleton = <Value>(
+  name: string,
+  valueFactory: () => Value
+): Value => {
+  const g = global as any;
+  g.__singletons ??= {};
+  g.__singletons[name] ??= valueFactory();
+  return g.__singletons[name];
+};
+```
 
 ```ts filename=app/utils/db.server.ts
 import { PrismaClient } from "@prisma/client";
 
-let db: PrismaClient;
+import { singleton } from "./singleton.server";
 
-declare global {
-  var __db__: PrismaClient | undefined;
-}
-
-// This is needed because in development we don't want to restart
-// the server with every change, but we want to make sure we don't
-// create a new connection to the DB with every change either.
-// In production, we'll have a single connection to the DB.
-if (process.env.NODE_ENV === "production") {
-  db = new PrismaClient();
-} else {
-  if (!global.__db__) {
-    global.__db__ = new PrismaClient();
-  }
-  db = global.__db__;
-  db.$connect();
-}
+// Hard-code a unique key, so we can look up the client when this module gets re-imported
+const db = singleton("prisma", () => new PrismaClient());
+db.$connect();
```

Comment:
Put the connect call within the callback, otherwise you'll call `$connect` on every file change which is probably not good.

Or just remove the connect call altogether. It's probably not necessary anyway.